src/test/regress/sql/subselect_gp_indexes.sql

```diff
@@ -0,0 +1,47 @@
+--
+-- Test correlated subquery in subplan with motion chooses correct scan type
+--
+-- Given distributed table
+create table subplan_motion_t
+(a bigint NOT NULL) DISTRIBUTED BY (a);
+-- with constraint
+alter table only subplan_motion_t add constraint users_choose_tidscan_pkey primary key (a);
+-- and some data
+insert into subplan_motion_t
+select gen from generate_series(1, 20000) gen;
+
+-- Check that Index Scan is not used
+select cte."b" from (select 7 as "b" ) cte
```

Comment:
There is no need to wrap column `b` with double quotes (it makes sense in a case of capital letters, for example).